core/Http/BadRequestException.php

```diff
@@ -9,10 +9,10 @@
 namespace Piwik\Http;
 
 
-class BadRequestException extends HttpCodeException
+class BadRequestException extends \Exception implements HttpCodeException
 {
     public function __construct($message)
     {
         parent::__construct($message, $code = 400);
     }
```

Comment:
We could also improve that one, so it's possible to overwrite the code if needed.
```suggestion
    public function __construct($message, $code = 400)
    {
        parent::__construct($message, $code);
    }
```